File: analytics_url.php
<?php
// analytics_url: http://<dominio>/analytics-atividade
header('Content-Type: application/json');
$input_json = file_get_contents('php://input');
$data = json_decode($input_json, true);
$activity_id = $data['activityID'] ?? null; // ID da Inven!RA.

if ($activity_id) {
    // Ação interna: Obter analytics de TODOS os alunos para esta activityID.
    // (Simulação de dados de resposta - o formato final deve corresponder ao esperado)
    $analytics_response = [
        [
            "inveniraStdID" => "1001",
            "quantAnalytics" => [
                ["name" => "Acedeu à atividade", "value" => true],
                ["name" => "Download documento 1", "value" => true],
                ["name" => "Evolução pela atividade ()", "value" => "33.3"]
            ],
            "qualAnalytics" => [
                "Student activity profile" => "https://ActivityProvider/?APAnID=11111111", // URL para o AP devolver a página de conteúdo qualitativo.
                "Actitivy Heat Map" => "https://ActivityProvider/?APAnID=21111111"
            ]
        ],
        // ... outros utilizadores
    ];
    
    echo json_encode($analytics_response);
} else {
    http_response_code(400);
    echo json_encode(["error" => "activityID em falta."]);
}
?>

File: analytics_url.py
# Arquivo: analytics_url.py
from flask import Blueprint, request, jsonify

analytics_bp = Blueprint('analytics', __name__)

@analytics_bp.route('/analytics-atividade', methods=['POST'])
def get_analytics_data():
    """
    Retorna os dados de analytics para uma activityID específica.
    """
    data = request.get_json()
    activity_id = data.get('activityID') if data else None

    if activity_id:
        # Simulação de resposta da Base de Dados
        analytics_response = [
            {
                "inveniraStdID": "1001",
                "quantAnalytics": [
                    {"name": "Acedeu à atividade", "value": True},
                    {"name": "Download documento 1", "value": True},
                    {"name": "Evolução pela atividade ()", "value": "33.3"}
                ],
                "qualAnalytics": {
                    "Student activity profile": f"{request.host_url}?APAnID=11111111",
                    "Actitivy Heat Map": f"{request.host_url}?APAnID=21111111"
                }
            }
        ]
        return jsonify(analytics_response)
    else:
        return jsonify({"error": "activityID em falta."}), 400